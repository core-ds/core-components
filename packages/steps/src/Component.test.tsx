import React from 'react';
import { render } from '@testing-library/react';

import { Steps } from './index';

describe('Steps', () => {
    describe('Snapshots tests', () => {
        it('should match snapshot', () => {
            expect(
                render(
                    <Steps>
                        <div>Шаг 1</div>
                        <div>Шаг 2</div>
                    </Steps>,
                ),
            ).toMatchSnapshot();
        });
    });

    describe('Classes tests', () => {
        it('should set custom class', () => {
            const className = 'custom-class';
            const { container } = render(
                <Steps className={className}>
                    <div>Шаг 1</div>
                    <div>Шаг 2</div>
                </Steps>,
            );

            expect(container.firstElementChild).toHaveClass(className);
        });

        it('should set `vertical` class', () => {
            const { container } = render(
                <Steps isVerticalAlign={true}>
                    <div>Шаг 1</div>
                    <div>Шаг 2</div>
                </Steps>,
            );

            expect(container.firstElementChild).toHaveClass('vertical');
        });
    });

    describe('Attributes tests', () => {
        it('should set `data-test-id` atribute', () => {
            const dataTestId = 'test-id';

            const { getByTestId } = render(
                <Steps dataTestId={dataTestId}>
                    <div>Шаг 1</div>
                    <div>Шаг 2</div>
                </Steps>,
            );

            expect(getByTestId(dataTestId).tagName).toBe('DIV');
        });
    });

    describe('Render tests', () => {
        test('should render with default active step', () => {
            const defaultActiveStep = 2;

            const { container } = render(
                <Steps defaultActiveStep={defaultActiveStep}>
                    <div>Шаг 1</div>
                    <div>Шаг 2</div>
                </Steps>,
            );

            const steps = container.querySelectorAll('.step');

            expect(steps[0]).toHaveClass('completed');
            expect(steps[defaultActiveStep - 1]).toHaveClass('selected');
        });

        test('should render ordered steps', () => {
            const { queryAllByText } = render(
                <Steps>
                    <div>Подготовка</div>
                    <div>Завершение</div>
                </Steps>,
            );

            const steps = queryAllByText(/\d/);

            expect(steps[0].innerHTML).toBe('1');
            expect(steps[1].innerHTML).toBe('2');
        });

        test('should render unordered steps', () => {
            const { container } = render(
                <Steps ordered={false}>
                    <div>Шаг 1</div>
                    <div>Шаг 2</div>
                    <div>Шаг 3</div>
                </Steps>,
            );

            const unorderedSteps = container.querySelectorAll('.checkbox > .dot');

            expect(unorderedSteps.length).toBe(3);
        });

        test('should unmount without errors', () => {
            const { unmount } = render(
                <Steps>
                    <div>Шаг 1</div>
                    <div>Шаг 2</div>
                </Steps>,
            );

            expect(unmount).not.toThrowError();
        });
    });
});
