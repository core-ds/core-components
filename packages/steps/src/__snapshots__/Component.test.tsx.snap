// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps Snapshots tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component"
      >
        <div
          class="step selected interactive"
          role="button"
          tabindex="0"
        >
          <div
            class="indicator interactive"
          >
            <div
              class="option"
            >
              1
            </div>
          </div>
          <div
            class="text interactive"
          >
            <div>
              Шаг 1
            </div>
          </div>
          <div
            class="dash"
          />
        </div>
        <div
          class="step interactive"
          role="button"
          tabindex="0"
        >
          <div
            class="indicator interactive"
          >
            <div
              class="option"
            >
              2
            </div>
          </div>
          <div
            class="text interactive"
          >
            <div>
              Шаг 2
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component"
    >
      <div
        class="step selected interactive"
        role="button"
        tabindex="0"
      >
        <div
          class="indicator interactive"
        >
          <div
            class="option"
          >
            1
          </div>
        </div>
        <div
          class="text interactive"
        >
          <div>
            Шаг 1
          </div>
        </div>
        <div
          class="dash"
        />
      </div>
      <div
        class="step interactive"
        role="button"
        tabindex="0"
      >
        <div
          class="indicator interactive"
        >
          <div
            class="option"
          >
            2
          </div>
        </div>
        <div
          class="text interactive"
        >
          <div>
            Шаг 2
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
